receiver/m365receiver/logs.go

```diff
@@ -0,0 +1,514 @@
+	}
+	if log.DLPExchangeMetaData != nil {
+		m.PutStr("dlp.exchange.message.id", log.DLPExchangeMetaData.MessageID)
+	}
+	if log.QuarantineSource != nil {
+		m.PutStr("quarantine.request_source", matchQuarantineSource(*log.QuarantineSource))
+	}
+	if log.DefenderFileSource != nil {
+		m.PutStr("defender.file.source_workload", matchSourceWorkload(*log.DefenderFileSource))
+	}
+	if log.CommCompliance != nil {
+		m.PutStr("communication_compliance.exchange.network.message.id", log.CommCompliance.NetworkMessageID)
+	}
+	if log.DataShareInvitation != nil {
+		m.PutStr("system_sync.data.share.id", log.DataShareInvitation.ShareID)
+	}
+	if log.DefenderFile != nil {
+		m.PutStr("defender.file.id", log.DefenderFile.DocumentID)
+		m.PutStr("defender.file.verdict", matchFileVerdict(log.DefenderFile.FileVerdict))
+	}
+	if log.SecurityAlertID != "" {
+		m.PutStr("security.alert.id", log.SecurityAlertID)
+		m.PutStr("security.alert.name", log.SecurityAlertName)
+	}
+	if log.ConnectorJobID != "" {
+		m.PutStr("compliance_connector.job.id", log.ConnectorJobID)
+		setAttributeIfNotEmpty(m, "compliance_connector.task.id", log.ConnectorTaskID)
+	}
+	if log.VivaGoalsUsername != "" {
+		m.PutStr("viva.username", log.VivaGoalsUsername)
+		setAttributeIfNotEmpty(m, "viva.organization.name", log.VivaGoalsOrgName)
+	}
+
+	recordType := matchRecordType(log.RecordType)
+	if recordType == "ThreatIntelligenceUrl" {
+		m.PutStr("defender.url.url", log.DefenderURL)
+	}
+	if recordType == "PowerBIAudit" {
+		m.PutStr("powerbi.app.name", log.PowerAppName)
+	}
+	if recordType == "OMEPortal" {
+		setAttributeIfNotEmpty(m, "encrypted_portal.message.id", log.EncryptedMessageID)
+	}
+	if recordType == "MicrosoftGraphDataConnectConsent" {
+		setAttributeIfNotEmpty(m, "graph.consent.app.id", log.MSGraphConsentAppID)
+	}
+
+	if log.AzureActor != nil {
+		slice := m.PutEmptySlice("azure.actors")
+		for _, r := range *log.AzureActor {
+			pair := slice.AppendEmpty().SetEmptyMap()
+			pair.PutStr("actor.id", r.ID)
+			pair.PutStr("actor.type", matchAzureUserType(r.Type))
+		}
+	}
+	if log.DLPPolicyDetails != nil {
+		slice := m.PutEmptySlice("dlp.policy_details")
+		for _, r := range *log.DLPPolicyDetails {
+			pair := slice.AppendEmpty().SetEmptyMap()
+			pair.PutStr("policy.id", r.PolicyID)
+			pair.PutStr("policy.name", r.PolicyName)
+		}
+	}
+
+	if log.DefenderEmail != nil {
+		slice := m.PutEmptySlice("defender.email.attachments")
+		for _, r := range *log.DefenderEmail {
+			pair := slice.AppendEmpty().SetEmptyMap()
+			pair.PutStr("file.name", r.FileName)
+		}
+	}
+
+}
+
+func matchUserType(x int) string {
+	switch x {
+	case 0:
+		return "Regular"
+	case 1:
+		return "Reserved"
+	case 2:
+		return "Admin"
+	case 3:
+		return "DcAdmin"
+	case 4:
+		return "System"
+	case 5:
+		return "Application"
+	case 6:
+		return "ServicePrincipal"
+	case 7:
+		return "CustomPolicy"
+	case 8:
+		return "SystemPolicy"
+	default:
+		return "impossible"
```

Comment:
For these switch statements, I think "unknown" might be better, they could be hit and show up in logs if something changes on the MS side.